compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -247,14 +260,69 @@ enum Scope<'a> {
         s: ScopeRef<'a>,
     },
 
+    TraitRefHackInner {
```

Comment:
Should add some docs. This is basically needed so we assign late bound vars to the actual trait red and not the list of trait refs/lifetimes